### PR TITLE
Examples and doc updates

### PR DESCRIPTION
**Pull request by mkinney**
Even more examples

It seems as I add examples, I find bugs or have questions about the operations.

Let me know if you want a break from these.  :-)

### CODE REVIEW

File: examples/apt.py [hunk 1]
```diff
@@ -3,10 +3,49 @@
 
 SUDO = True
 
-if host.fact.linux_name in ['Debian', 'Ubuntu']:
+# Note: Using linux_distribution fact so running from docker
+# will show valid name, otherwise could just use host.fact.linux_name
+linux_distribution = host.fact.linux_distribution
```
Comment (Fizzadar) [inline examples/apt.py]: This could all be written as:

```py
code_name = host.fact.linux_distribution['release_meta'].get('DISTRIB_CODENAME')

if host.fact.linux_name...
```

The `linux_name` fact actually uses `linux_distribution` under the hood - so this won't cause any double load.

File: examples/apt.py [hunk 2]
```diff
@@ -3,10 +3,49 @@
...
+    )
+
+    apt.key(
+        {'Install VirtualBox key'},
+        'https://www.virtualbox.org/download/oracle_vbox_2016.asc',
+    )
+
+    apt.repo(
+        {'Install VirtualBox repo'},
+        'deb https://download.virtualbox.org/virtualbox/debian {{ code_name }} contrib',
```
Comment (Fizzadar) [inline examples/apt.py]: This string should use f-strings or `.format` for `code_name`, rather than a jinja2 template variable. The inline jinja2-style variables are no longer required and will be deprecated/maybe removed at some point!

File: examples/git.py [hunk 3]
```diff
@@ -1,9 +1,31 @@
-from pyinfra.modules import git
+from pyinfra import host
+from pyinfra.modules import apk, apt, files, git
 
 SUDO = True
 
+linux_name = host.fact.linux_distribution.get('name', '')
```
Comment (Fizzadar) [inline examples/git.py]: This can just be `host.fact.linux_name`.

File: examples/apt.py (hunk 1 above)
Comment (mkinney) [reply examples/apt.py]: Thanks

File: examples/apt.py (hunk 2 above)
Comment (mkinney) [reply examples/apt.py]: Thanks

File: examples/git.py (hunk 3 above)
Comment (mkinney) [reply examples/git.py]: Ok. Thanks.

**Comment by mkinney** [comment]
I'm ok if you merge this one, if you are ok with changes.

File: pyinfra/modules/yum.py [hunk 4]
```diff
@@ -107,10 +109,11 @@ def rpm(state, host, source, present=True):
 
     .. code:: python
 
+        linux_major = host.fact.linux_distribution['release_meta'].get('VERSION_ID')
         yum.rpm(
            {'Install EPEL rpm to enable EPEL repo'},
            'https://dl.fedoraproject.org/pub/epel/epel-release-latest-'
-           '{{  host.fact.linux_distribution.major }}.noarch.rpm',
+           '{}.noarch.rpm'.format(linux_major),
```
Comment (Fizzadar) [inline pyinfra/modules/yum.py]: I'm not sure why this change is needed? `host.fact.linux_distribution.major` should return the same value?

Comment (mkinney) [reply pyinfra/modules/yum.py]: Sorry. I misunderstood the jinja2 comment above. Reverted.
